src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPAuthenticationScheme.cs

```diff
@@ -71,23 +73,36 @@ public string FormatAccessToken(MsalAccessTokenCacheItem msalAccessTokenCacheIte
             var payload = new JObject(
                 new JProperty(PoPClaimTypes.At, msalAccessTokenCacheItem.Secret),
                 new JProperty(PoPClaimTypes.Ts, (long)(DateTime.UtcNow - s_jwtBaselineTime).TotalSeconds ),
-                new JProperty(PoPClaimTypes.HttpMethod, _httpRequestMessage.Method.ToString()),
-                new JProperty(PoPClaimTypes.Host, _httpRequestMessage.RequestUri.Host),
-                new JProperty(PoPClaimTypes.Path, _httpRequestMessage.RequestUri.AbsolutePath),
+                new JProperty(PoPClaimTypes.HttpMethod, _popAuthenticationConfiguration.HttpMethod?.ToString()),
+                new JProperty(PoPClaimTypes.Host, _popAuthenticationConfiguration.RequestUri.Host),
+                new JProperty(PoPClaimTypes.Path, _popAuthenticationConfiguration.RequestUri.AbsolutePath),
                 new JProperty(PoPClaimTypes.Nonce, CreateNonce()),
                 new JProperty(PoPClaimTypes.Cnf, new JObject(
                     new JProperty(PoPClaimTypes.JWK, popAssertion))));
 
             string popToken =  CreateJWS(payload.ToString(Json.Formatting.None), header.ToString(Json.Formatting.None));
 
             // For POP, we can also update the HttpRequest with the authentication header
-            _httpRequestMessage.Headers.Authorization = new AuthenticationHeaderValue(
+            _popAuthenticationConfiguration.SetPopHttpRequestHeaders(new AuthenticationHeaderValue(
                 AuthorizationHeaderPrefix,
-                popToken);
+                popToken));
 
             return popToken;
         }
 
+        private string GetHeaderAlgorithm()
+        {
+            var JWK = JObject.Parse(_popAuthenticationConfiguration.PopCryptoProvider.CannonicalPublicKeyJwk);
+
+            //Need to update header to signal EC key is being used
+            if (JWK[JsonWebKeyParameterNames.Kty].ToString() == JsonWebKeyParameterNames.EC)
+            {
+                return JsonWebTokenConstants.Algorithms.EdcSha256;
```

Comment:
Need to add support for other elliptic curve key sizes here. currently only 256 is supported. 
@bgavrilMS

Reply:
supported values
https://docs.microsoft.com/en-us/azure/key-vault/keys/about-keys#signverify